plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/WebXmlValidatorTest.java

```diff
@@ -16,73 +16,43 @@
-      + " version='3.1'></web-app>";
-  private static final String ELEMENT_MESSAGE =
-      "App Engine Standard does not support this servlet version";
-  private static final String SERVLET_MARKER =
-      "com.google.cloud.tools.eclipse.appengine.validation.servletMarker";
-  private static IResource resource;
-  private static IProject project;
-  
-  @ClassRule public static TestProjectCreator projectCreator = new TestProjectCreator();
-  
-  @BeforeClass
-  public static void setUp() throws CoreException {
-    project = projectCreator.getProject();
-    createFolders(project, new Path("src/main/webapp/WEB-INF"));
-    IFile webXml = project.getFile("src/main/webapp/WEB-INF/web.xml");
-    webXml.create(new ByteArrayInputStream(new byte[0]), true, null);
-    resource = webXml;
-  }
-  
-  
-  @After
-  public void tearDown() throws CoreException {
-    if (resource != null) {
-      resource.deleteMarkers(IMarker.PROBLEM, true, IResource.DEPTH_ZERO);
-    }
-  }
-
-  @Test
-  public void testValidate_withWrongVersion()
-      throws IOException, CoreException, ParserConfigurationException {
-    byte[] bytes = XML.getBytes(StandardCharsets.UTF_8);
-    WebXmlValidator validator = new WebXmlValidator();
-    validator.validate(resource, bytes);
-    IMarker[] markers = resource.findMarkers(SERVLET_MARKER, true, IResource.DEPTH_ZERO);
-    assertEquals(1, markers.length);
-    assertEquals(ELEMENT_MESSAGE, (String) markers[0].getAttribute(IMarker.MESSAGE));
-  }
+ 
+  IFile file = Mockito.mock(IFile.class);
   
-  private static void createFolders(IContainer parent, IPath path) throws CoreException {
-    if (!path.isEmpty()) {
-      IFolder folder = parent.getFolder(new Path(path.segment(0)));
-      folder.create(true,  true,  null);
-      createFolders(folder, path.removeFirstSegments(1));
-    }
-  }
+ @Test
+ public void testValidate() throws CoreException, IOException, ParserConfigurationException {
+   String markerId = "com.google.cloud.tools.eclipse.appengine.validation.servletMarker";
+   when(file.createMarker(markerId)).thenReturn(Mockito.mock(IMarker.class));
+   String xml = 
+       "<web-app xmlns='http://xmlns.jcp.org/xml/ns/javaee' version='3.1'></web-app>";
+   byte[] bytes = xml.getBytes(StandardCharsets.UTF_8);
+   WebXmlValidator validator = new WebXmlValidator();
+   validator.validate(file, bytes);
+   Mockito.verify(file, Mockito.times(1)).createMarker(markerId);
+ }
+ 
+ @Test
+ public void testValidate_noMarkers()
+     throws CoreException, IOException, ParserConfigurationException {
+   String xml = 
+       "<web-app xmlns='http://java.sun.com/xml/ns/javaee' version='2.5'></web-app>";
+   byte[] bytes = xml.getBytes(StandardCharsets.UTF_8);
+   WebXmlValidator validator = new WebXmlValidator();
+   validator.validate(file, bytes);
+   String markerId = "com.google.cloud.tools.eclipse.appengine.validation.servletMarker";
+   Mockito.verify(file, Mockito.times(0)).createMarker(markerId);
+ }
+ 
```

Comment:
Should we have a test that web.xml validation _doesn't_ happen on non-Dynamic Web Facet projects?